tests/integration/helpers.py

```diff
@@ -217,5 +217,55 @@ async def secondary_mongo_uris_with_sync_delay(ops_test: OpsTest, rs_status_data
 
 
 def generate_collection_id() -> str:
-    new_id = "".join(choices(ascii_lowercase + digits, k=4)).replace("_", "")
+    """Generates a short and random Mongodb collection id."""
+    # force updating the seed to the current time
+    seed(datetime.now())
+
+    new_id = "".join(choices(ascii_lowercase + digits, k=4))
     return f"collection_{new_id}"
+
+
+def get_latest_unit_id(ops_test: OpsTest) -> int:
+    units = ops_test.model.applications[APP_NAME].units
+
+    unit_ids = [unit.name.split("/")[-1] for unit in units]
+
+    return int(max(unit_ids))
+
+
+async def get_current_storage_info(ops_test: OpsTest) -> SimpleNamespace:
+    """Returns the info of the storage state of the latest juju unit.
+
+    Returns:
+        SimpleNamespace(
+            unit_id=latest_juju_unit_id,
+            k8s_volume_id=corresponding K8s pvc id
+        )
+    """
+
+    def get_unit_volume_ids(full_storage_obj) -> dict:
```

Comment:
seems to return only one val
```suggestion
    def get_unit_volume_id(full_storage_obj) -> dict:
```

Reply:
It returns a dictionary of "unit_id -> pvc_id" (you'll notice a nested loop)